### ISSUE TITLE
RFC-0204/TariScriptExamples references deprecated range proof rewinding

### ISSUE THREAD

**Issue by AaronFeickert** [tari-project/rfcs#83]
The one-sided payment example in [RFC-0204](https://rfc.tari.com/RFC-0204_TariScriptExamples.html#one-sided-payment) is out of date. It references the use of [RFC-0180](https://rfc.tari.com/RFC-0180_BulletproofRewinding.html) Bulletproof range proof rewinding. In addition to Bulletproofs being superseded by Bulletproofs+ for range proving and mask recovery, value communication is now done using a separate encryption.

This should be changed to reflect the [work-in-progress PR](https://github.com/tari-project/rfcs/pull/82) for Bulletproofs+ range proving.

**Comment by AaronFeickert** [comment]
This is about to become even more out of date than before.

A [draft PR](https://github.com/tari-project/tari/pull/5372) suggests making range proofs optional, and encrypting commitment values and masks together. Once it is merged, any references to range proof rewinding in RFC-0204 will be inaccurate. While rewinding will still be possible in the Bulletproofs+ library, the default client will no longer support it.

Another [draft PR](https://github.com/tari-project/rfcs/pull/98) suggests a change to interactive outputs to better support scanning without access to a wallet's secret key. Once it is implemented, the discussion of such outputs in RFC-0204 will be inaccurate.
